examples/minimum-viable-router/src/main.rs

```diff
@@ -0,0 +1,62 @@
+use route_rs_packets::EthernetFrame;
+use route_rs_runtime::link::primitive::{InputChannelLink, OutputChannelLink};
+use route_rs_runtime::link::{LinkBuilder, PacketStream, TokioRunnable};
+use tokio::task::JoinHandle;
+
+mod route;
+
+fn main() {
+    let mut runtime = tokio::runtime::Builder::new()
+        .threaded_scheduler()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    run_route(vec![], vec![], &mut runtime, route::Route::new())
+}
+
+// TODO: Move this to route-rs-runtime
+fn run_route(
+    ingressors: Vec<crossbeam::Receiver<EthernetFrame>>,
+    egressors: Vec<crossbeam::Sender<EthernetFrame>>,
+    runtime: &mut tokio::runtime::Runtime,
+    route: route::Route,
+) {
+    let mut all_runnables = vec![];
+
+    let (_, upstreams): (
+        Vec<Vec<TokioRunnable>>,
+        Vec<Vec<PacketStream<EthernetFrame>>>,
+    ) = ingressors
+        .into_iter()
+        .map(|receiver| InputChannelLink::new().channel(receiver).build_link())
+        .unzip();
+
+    let (mut route_runnables, downstreams) = route
+        .ingressors(upstreams.into_iter().flatten().collect())
+        .build_link();
+
+    all_runnables.append(&mut route_runnables);
+
+    let egressor_inputs = downstreams.into_iter().zip(egressors.into_iter());
+
+    let (egress_runnables, _): (Vec<Vec<TokioRunnable>>, Vec<Vec<PacketStream<_>>>) =
+        egressor_inputs
+            .map(|(ingressor, sender)| {
+                OutputChannelLink::new()
+                    .ingressor(ingressor)
+                    .channel(sender)
+                    .build_link()
+            })
+            .unzip();
+
+    all_runnables.append(&mut egress_runnables.into_iter().flatten().collect());
+
```

Comment:
All this building needs to happen inside the block_on block, otherwise you can get initialization failures.  The block on provides the runtime context. It will be a rare error, but if people have external timer links and stuff it won't run.

Reply:
Especially because if Nathans links are initialized outside of the "router", those will need the tokio context active since they need to access the reactor and executor threads of tokio.

Reply:
The way I envisioned this is that the `ingressors` and `egressors` would be replaced with the socket objects and we'd still assemble the links inside of `run_route`. Probably given that we should have a less generic name for this method, since it bakes in the use of `af_packet`.

But it should be relatively okay to just extend the `block_on` around the rest of this method body.

Reply:
yeah, that makes sense to me. The block_on thing is annoying,  Another annoyance is that the `route` object needs to be also built inside a tokio context.  So we should probably change that arguments to something like `route: Fn() -> Link`, then we can call `let router = route::new();` inside the tokio `block_on`.  For an example of this, I did it inside the prototype run harness in the runtime, under `utils`

Reply:
The Fn doesn't have to return a `Link` either, could be some other struct, but returning the `Link` makes some sense from our composability story.

Reply:
Do we have to initialize the `route` object fully inside the `block_on`? Or can we just `build_link()` in there. Keeping it outside allows the implementor to pass additional setup fields in `main()` if desired.

Reply:
Only the "Link" objects need to be initialized inside the block_on, so yes we could pre-init all the builder syntax stuff outside the context, then pass the object that you are about to build_link() on into the context, and call build_link.   I think that is a pretty good idea.